flink-table-store-core/src/main/java/org/apache/flink/table/store/file/data/DataFileMetaSerializer.java

```diff
@@ -48,27 +42,25 @@ public RowData toRow(DataFileMeta meta) {
                 StringData.fromString(meta.fileName()),
                 meta.fileSize(),
                 meta.rowCount(),
-                meta.minKey(),
-                meta.maxKey(),
-                keyStatsArraySerializer.toRow(meta.keyStats()),
-                valueStatsArraySerializer.toRow(meta.valueStats()),
+                serializeBinaryRow(meta.minKey()),
```

Comment:
Current implementation will copy `BinaryRowData` into a new byte array. Can we keep `BinaryRowData` and read its underlying byte array directly when serializing? This can be applied not only to `minKey` and `maxKey`, but also to `partitionKey` and even `BinaryTableStats`. Also by storing a `BinaryRowData` in the object we can read its fields directly if we know the types, which simplifies the implementation a lot.

Reply:
It is difficult:
- First, the row serializer is dependent on flink, if we want to have this logic, we need to write a serializer ourselves.
- Second, avro serialization does not use serializer, it is to get rowData to convert itself.